src/index.test.ts

```diff
@@ -351,6 +351,42 @@ describe('post()', () => {
     // assert error is empty
     expect(error).toBe(undefined);
   });
+
+  it('request body type when optional', async() => {
+    fetchMocker.mockResponse(() => ({ status: 201, body: '{}' }));
+    const client = createClient<paths>();
+
+    // expect error on wrong body type
+    // @ts-expect-error
+    await client.post('/post/optional', { body: { error: true } })
```

Comment:
🎉 Awesome! These types of inverse tests are very helpful (sometimes even more helpful than the “assert no error” tests.